src/script/lua_api/l_vmanip.cpp

```diff
@@ -111,14 +110,20 @@ int LuaVoxelManip::l_set_data(lua_State *L)
 
 int LuaVoxelManip::l_write_to_map(lua_State *L)
 {
-	GET_ENV_PTR;
-
 	LuaVoxelManip *o = checkObject<LuaVoxelManip>(L, 1);
 	bool update_light = !lua_isboolean(L, 2) || readParam<bool>(L, 2);
 
 	if (o->vm->isOrphan())
 		return 0;
 
+	// In mapgen code the vmanip cannot be written manually
+	if (o->is_mapgen_vm) {
+		log_deprecated(L, "write_to_map called for a mapgen VoxelManip object");
+		return 0;
+	}
```

Comment:
https://github.com/minetest/minetest/blob/669284c7a2ff036c151334f77041866f235e6642/src/script/lua_api/l_vmanip.cpp#L130

The `o->is_mapgen_vm` part could be removed.

Reply:
Anyway, this check seems to break regular / synchronous Lua map generators (using `minetest.register_on_generated`).

An example:

```lua
minetest.set_mapgen_setting("mg_name", "singlenode", true)

minetest.register_on_generated(function(minp, maxp, blockseed)
    local vm, vm_minp, vm_maxp = minetest.get_mapgen_object("voxelmanip")
    local vm_area = VoxelArea:new{MinEdge = vm_minp, MaxEdge = vm_maxp}
    local vm_data = vm:get_data()

    local air = minetest.get_content_id("air")
    local stone = minetest.get_content_id("mapgen_stone")

    for x = minp.x, maxp.x do
    for y = minp.y, maxp.y do
    for z = minp.z, maxp.z do
        local index = vm_area:index(x, y, z)

        if y == -2 then
            vm_data[index] = stone
        elseif x % 2 == 0 and z % 2 == 0 then
            vm_data[index] = stone
        else
            vm_data[index] = air
        end
    end
    end
    end

    vm:set_data(vm_data)
    vm:write_to_map()
end)
```

With this PR, this snippet no longer generates anything. If I remove this block of code, it works again:

https://github.com/minetest/minetest/blob/669284c7a2ff036c151334f77041866f235e6642/src/script/lua_api/l_vmanip.cpp#L119-L123

Reply:
oops, fixed